--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -24,11 +24,14 @@ endif()
 file(GLOB SRCS ./*.cpp ./*.cu ./*.h* ${LIBSGM_INCLUDE_DIR}/*.h*)
 add_library(${PROJECT_NAME} ${SGM_LIB_TYPE})
 target_sources(${PROJECT_NAME} PRIVATE ${SRCS})
-target_include_directories(${PROJECT_NAME} PRIVATE ${LIBSGM_INCLUDE_DIR} $<$<BOOL:${BUILD_OPENCV_WRAPPER}>:${OpenCV_INCLUDE_DIRS}>)
+target_include_directories(${PROJECT_NAME} 
+    PUBLIC
+        $<BUILD_INTERFACE:${LIBSGM_INCLUDE_DIR}>
+        $<INSTALL_INTERFACE:include>
+    PRIVATE
+        $<$<BOOL:${BUILD_OPENCV_WRAPPER}>:${OpenCV_INCLUDE_DIRS}>
+)
 target_compile_features(${PROJECT_NAME} PRIVATE cxx_std_17)
 target_link_libraries(${PROJECT_NAME} PUBLIC CUDA::cudart $<$<BOOL:${BUILD_OPENCV_WRAPPER}>:${OpenCV_LIBS}>)
-set_target_properties(${PROJECT_NAME} PROPERTIES INTERFACE_INCLUDE_DIRECTORIES ${LIBSGM_INCLUDE_DIR})
-
 target_compile_options(${PROJECT_NAME} PRIVATE
 $<$<COMPILE_LANG_AND_ID:CXX,GNU>:-Wall -O3>
 $<$<COMPILE_LANG_AND_ID:CXX,Clang>:-Wall -O3>
